### PR TITLE
Add support for architecture sparc-v8 and soc bm3803.

### PR DESCRIPTION
**Pull request by GFWisshit**
## 拉取/合并请求描述：(PR description)

[Add support for architecture sparc-v8 and soc bm3803]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [x] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/bm3803/applications/board.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+ * File      : board.c
```
Comment (BernardXiong) [inline bsp/bm3803/applications/board.c:2]: Please use new copyright file header with Apache License v2.0

File: bsp/bm3803/drivers/SConscript [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Import('RTT_ROOT')
+Import('rtconfig')
+from building import *
+
+cwd     = GetCurrentDir()
+src	= Glob('*.c')
+
+# remove no need file.
+if GetDepend('RT_USING_LWIP') == False:
+    src_need_remove = ['dm9000.c'] # need remove file list.
```
Comment (BernardXiong) [inline bsp/bm3803/drivers/SConscript:11]: Does bm3803 support ETH with DM9000? Where is the DM9000 driver?

Comment (GFWisshit) [reply bsp/bm3803/drivers/SConscript:11]: 你们的cla我填错用户名了
这份代码的结构是搬你们beaglebone bsp写的

Comment (GFWisshit) [reply bsp/bm3803/drivers/SConscript]: 改完了，可以再看一下。

File: bsp/bm3803/applications/board.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
```
Comment (BernardXiong) [inline bsp/bm3803/applications/board.c:1]: Please use correct format:

```c
/*
 * Copyright (c) 2020, your team information.
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Change Logs:
 * Date           Author       Notes
 * your logs information....
 */
```

File: libcpu/sparc-v8/common/SConscript [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+from building import *
+
+Import('rtconfig')
+
+cwd     = GetCurrentDir()
+src     = Glob('*.c') + Glob('*.cpp')
+CPPPATH = [cwd]
+
+if rtconfig.PLATFORM == 'armcc':
```
Comment (BernardXiong) [inline libcpu/sparc-v8/common/SConscript:11]: There is no 'armcc' and 'iar' compiler for SPARC.

Comment (GFWisshit) [reply libcpu/sparc-v8/common/SConscript]: 改完了，可以再看一下。

**Comment by BernardXiong** [comment]
No README.md file?

**Comment by GFWisshit** [comment]
> No README.md file?

暂时没有，以后可能会补上，这是航空航天专用的抗辐照处理器。

**Comment by BernardXiong** [comment]
> > No README.md file?
> 
> 暂时没有，以后可能会补上，这是航空航天专用的抗辐照处理器。

Please provide README.md.

**Comment by GFWisshit** [comment]
> > > No README.md file?
> > 
> > 
> > 暂时没有，以后可能会补上，这是航空航天专用的抗辐照处理器。
> 
> Please provide README.md.

添加了一份README.md。

**Pull request merged**